### PR TITLE
Implement closure_time for GNTW::batchTransfer

### PR DESCRIPTION
**Pull request by Krigpl**
`batchTransfer` is supposed to handle all payments that happened before some `closure_time`. To avoid race condition between calling `batchTransfer` and adding new payments we look at the payments' timestamps and choose only those older than some specified threshold (currently set to 10 seconds).

### CODE REVIEW

File: golem/ethereum/paymentprocessor.py [hunk 1]
```diff
@@ -573,10 +575,17 @@ def sendout(self):
                 log.info("Next sendout in {} s".format(self.deadline - now))
                 return False
 
-            payments = self._awaiting
-            self._awaiting = []
+            closure_time = now - self.CLOSURE_TIME_DELAY
+
+            payments = \
+                [p for p in self._awaiting if p.timestamp <= closure_time]
+            print(self._awaiting[0].timestamp)
```
Comment (jiivan) [inline golem/ethereum/paymentprocessor.py]: :wastebasket: `print`

File: tests/golem/transactions/ethereum/test_paymentprocessor.py [hunk 2]
```diff
@@ -726,9 +730,58 @@ def test_batch_transfer(self):
...
+        self.pp.deadline = 0
+        tx = mock.Mock()
+        tx_hash = '0xdead'
+        tx.hash = decode_hex(tx_hash)
+        self.client.send.return_value = tx_hash
+        self.token.batch_transfer.return_value = tx
+
+        closure_time = 2
+        time_value = closure_time + self.pp.CLOSURE_TIME_DELAY
+        with patch('time.time', return_value=time_value):
```
Comment (jiivan) [inline tests/golem/transactions/ethereum/test_paymentprocessor.py]: `freezegun`

File: golem/ethereum/paymentprocessor.py [hunk 3]
```diff
@@ -574,10 +576,16 @@ def sendout(self):
...
-            payments = self._awaiting
-            self._awaiting = []
+            closure_time = now - self.CLOSURE_TIME_DELAY
+
+            payments = \
+                [p for p in self._awaiting if p.processed_ts <= closure_time]
+            if not payments:
+                return False
+            self._awaiting = \
+                [p for p in self._awaiting if p.processed_ts > closure_time]
```
Comment (jiivan) [inline golem/ethereum/paymentprocessor.py:592]: Is it a race condition here on `self._awaiting`?

Comment (Krigpl) [reply golem/ethereum/paymentprocessor.py:592]: This is inside a lock

**Review by jiivan — APPROVED** [review]

**Pull request merged**